src/base/static/state/ducks/places.js

```diff
@@ -44,7 +44,13 @@ const INITIAL_STATE = [];
 export default function reducer(state = INITIAL_STATE, action) {
   switch (action.type) {
     case SET_PLACES:
-      return action.payload;
+      return action.payload.reduce((memo, newPlace) => {
+        return !memo.find(place => newPlace.id === place.id)
```

Comment:
I guess it's a lot of extra work to check each model as it's added to the store, but this action will run at most twice so I hope it's ok.